packages/cli/src/commands/setup/vite/vite.js

```diff
@@ -0,0 +1,154 @@
+import fs from 'fs'
+import path from 'path'
+
+import toml from '@iarna/toml'
+import chalk from 'chalk'
+import { Listr } from 'listr2'
+
+import { addWebPackages } from '@redwoodjs/cli-helpers'
+import { getConfigPath } from '@redwoodjs/internal/dist/paths'
+import { errorTelemetry } from '@redwoodjs/telemetry'
+
+import { getPaths, transformTSToJS, writeFile } from '../../../lib'
+import c from '../../../lib/colors'
+import { isTypeScriptProject } from '../../../lib/project'
+
+export const command = 'vite'
+export const description =
+  '[Experimental] Configure the web side to use Vite, instead of Webpack'
+export const builder = (yargs) => {
+  yargs.option('force', {
+    alias: 'f',
+    default: false,
+    description: 'Overwrite existing configuration',
+    type: 'boolean',
+  })
+  yargs.option('verbose', {
+    alias: 'v',
+    default: false,
+    description: 'Print more logs',
+    type: 'boolean',
+  })
+  yargs.option('add-package', {
+    default: true,
+    description:
+      'Allows you to skip adding the @redwoodjs/vite package. Useful for testing',
+    type: 'boolean',
+  })
+}
+
+export const handler = async ({ force, verbose, addPackage }) => {
+  const ts = isTypeScriptProject()
+  const tasks = new Listr(
+    [
+      {
+        title: 'Confirmation',
+        task: async (_ctx, task) => {
+          const confirmation = await task.prompt({
+            type: 'Confirm',
+            message: 'Vite support is experimental. Continue?',
+          })
+
+          if (!confirmation) {
+            throw new Error('User aborted')
+          }
+        },
+      },
+      {
+        title: 'Adding vite.config.js...',
+        task: () => {
+          const viteConfigPath = `${getPaths().web.base}/vite.config.${
+            ts ? 'ts' : 'js'
+          }`
+
+          const templateContent = fs.readFileSync(
+            path.resolve(__dirname, 'templates', 'vite.config.ts.template'),
+            'utf-8'
+          )
+
+          const viteConfigContent = ts
+            ? templateContent
+            : transformTSToJS(viteConfigPath, templateContent)
+
+          return writeFile(viteConfigPath, viteConfigContent, {
+            overwriteExisting: force,
+          })
+        },
+      },
+      {
+        title: 'Adding Vite bundler flag to redwood.toml...',
+        task: (_ctx, task) => {
+          const redwoodTomlPath = getConfigPath()
+          const config = toml.parse(fs.readFileSync(redwoodTomlPath, 'utf-8'))
+
+          if (config.web.bundler !== 'vite') {
+            config.web.bundler = 'vite'
+
+            writeFile(redwoodTomlPath, toml.stringify(config), {
+              overwriteExisting: true, // redwood.toml always exists
+            })
```

Comment:
Seems like this step ruins the formatting and deletes the comments:

```toml
[web]
title = "Redwood App"
port = 8_910
apiUrl = "/.redwood/functions"
includeEnvironmentVariables = [ ]
bundler = "vite"

[api]
port = 8_911

[browser]
open = true

```

Reply:
Yeah noticed this too, but don't intend on solving right now.

Reply:
Actually fixed it with a simple string replace!